lib/api/apiUtils/object/objectLockHelpers.js

```diff
@@ -0,0 +1,64 @@
+const { errors } = require('arsenal');
+/**
+ * Calculates retain until date for the locked object version
+ * @param {object} retention - includes days or years retention period
+ * @return {object} the date until the object version remains locked
+ */
+function calculateRetainUntilDate(retention) {
+    const { days, years } = retention;
+    const date = new Date();
+    // Calculate the number of days to retain the lock on the object
+    const retainUntilDays = days || years * 365;
+    const retainUntilDate
+        = date.setDate(date.getDate() + parseInt(retainUntilDays, 10));
+    return new Date(retainUntilDate);
+}
+
+/**
+ * Validates object lock headers
+ * @param {object} bucket - bucket metadata
+ * @param {object} headers - request headers
+ * @param {object} log - the log request
+ * @return {object} - object with error if validation fails
+ */
+function validateHeaders(bucket, headers, log) {
+    const validation = {};
+    const bucketObjectLockEnabled = bucket.isObjectLockEnabled();
+    const objectLockMode = headers['x-amz-object-lock-mode'];
+    const objectLockDate = headers['x-amz-object-lock-retain-until-date'];
+    const objectLegalHold
+        = headers['x-amz-object-lock-legal-hold'];
+    // If retention headers or legal hold header present but
+    // object lock is not enabled on the bucket return error
+    if ((objectLockDate || objectLockMode || objectLegalHold)
+        && !bucketObjectLockEnabled) {
+        log.trace('bucket is missing ObjectLockConfiguration');
+        validation.error = errors.InvalidRequest.customizeDescription(
+            'Bucket is missing ObjectLockConfiguration');
+    }
+    if ((objectLockMode || objectLockDate) &&
+        !(objectLockMode && objectLockDate)) {
+        validation.error = errors.InvalidArgument.customizeDescription(
+            'x-amz-object-lock-retain-until-date and ' +
+            'x-amz-object-lock-mode must both be supplied'
+        );
+    }
+    const validModes = new Set(['GOVERNANCE', 'COMPLIANCE']);
+    const validLegalHolds = new Set(['ON', 'OFF']);
+    if ((objectLockMode && !validModes.has(objectLockMode))
+        || (objectLegalHold && !validLegalHolds.has(objectLegalHold))) {
+        validation.error = errors.InvalidArgument.customizeDescription(
+            'Unknown wormMode directive');
```

Comment:
Is this the right error message? wormMode?

Reply:
it actually is. It's a weird one for sure

Reply:
ok, just wanted to make sure...

Reply:
@ilkescality idk if this is true for put object, but when i was testing for intitiateMPU, I found that AWS doesn't actually return an error if the legalHold value isn't `ON` or `OFF`. I think it's better that we return an error so people expecting their objects to have legal hold know their setting didn't work, but I think it should be a different error than the mode one.  'Unknown legal hold value' or something like that

Reply:
@dora-korpar I just noticed this comment of yours, I think that's a good idea to return a more meaningful error such as 'Unknown legal hold value' for the case of the invalid legal hold values (unlike aws which seems to be silently failing which is not ideal). I will take care of it.